textFit.js

```diff
@@ -139,33 +143,89 @@
+        innerSpan.scrollHeight >= parseInt(window.getComputedStyle(innerSpan)['font-size'], 10) * 2) {
       multiLine = true;
     }
 
     // If we're not treating this as a multiline string, don't let it wrap.
     if (!multiLine) {
       el.style['white-space'] = 'nowrap';
     }
+    var startingSize = innerSpan.style.fontSize;
 
     low = settings.minFontSize;
     high = settings.maxFontSize;
-
     // Binary search for highest best fit
     var size = low;
     while (low <= high) {
-      mid = (high + low) >> 1;
-      innerSpan.style.fontSize = mid + 'px';
-      if(innerSpan.scrollWidth <= originalWidth && (settings.widthOnly || innerSpan.scrollHeight <= originalHeight)){
+      mid = parseFloat(((high + low) / 2).toFixed(2));
+      // mid = (high + low) >> 1;
+      // console.table('low: ' + low + ' mid: ', + mid +  ' high: ' + high);
+      innerSpan.style.fontSize = mid + settings.fontUnit;
+      if(innerSpan.scrollWidth <= originalWidth && (settings.widthOnly || innerSpan.scrollHeight <= originalHeight)) {
         size = mid;
-        low = mid + 1;
+        low = mid + settings.fontChangeSize;
       } else {
-        high = mid - 1;
+        high = mid - settings.fontChangeSize;
       }
       // await injection point
     }
+    if (startingSize !== size) {
+      console.log('textFit font changed size: ', size + settings.fontUnit)
+    }
     // found, updating font if differs:
-    if( innerSpan.style.fontSize != size + 'px' ) innerSpan.style.fontSize = size + 'px';
+    if( innerSpan.style.fontSize != size + settings.fontUnit ) innerSpan.style.fontSize = size + settings.fontUnit;
+
+    // add the required CSS in order to stop overflows
+    if (settings.maxLines || settings.stopOverflow) {
+      if (!document.getElementById("overflowStyleSheet")) {
+        var style = [
+          ".overflow > span {",
+            "overflow: hidden;",
+          "}"].join("");
+    
+        var css = document.createElement("style");
+        css.type = "text/css";
+        css.id = "overflowStyleSheet";
+        css.innerHTML = style;
+        document.body.appendChild(css);
+    }
+
+    el.classList.remove('overflow');
+
+    if (settings.maxLines) {
+      if (Number.isInteger(settings.maxLines)) {
+        const computedStyle = window.getComputedStyle(innerSpan);
+        // Checking if line-height is not set and has a normal value set by default
+        const isNormal = computedStyle.getPropertyValue('line-height') == 'normal';
+        // Getting the font-size of an element that will help us calculate the line-height
+        const fontSize = parseFloat(computedStyle.getPropertyValue('font-size'));
+        // If line-height is normal, we multiply the element's font-size with 1.14
+        if ( isNormal ) innerSpan.style.lineHeight = (fontSize * 1.14) + 'px';
+        const lineHeight = parseFloat(computedStyle.getPropertyValue('line-height'));
```

Comment:
```suggestion
        const lineHeightFloat = parseFloat(computedStyle.getPropertyValue('line-height')); 
        const lineHeight = Number.isFinite(lineHeightFloat) ? lineHeightFloat : parseFloat(computedStyle.getPropertyValue('font-size') * 1.14;
```

Reply:
Basically can remove the double-read here and simplify the logic a bit by simply assuming `fontSize * 1.14` if lineHeight is unreadable